src/libcrun/cgroup-internal.h

```diff
@@ -76,8 +79,22 @@ int libcrun_cgroup_pause_unpause_path (const char *cgroup_path, const bool pause
 static inline uint64_t
 convert_shares_to_weight (uint64_t shares)
 {
-  /* convert linearly from 2-262144 to 1-10000.  */
-  return (1 + ((shares - 2) * 9999) / 262142);
+  double l, exponent;
+
+  /* The value of 0 means "unset".  */
+  if (shares == 0)
+    return 0;
+  if (shares < 2)
+    return 1;
+  if (shares > 262144)
+    return 10000;
```

Comment:
nit: can use <= and >= here to avoid costly calculation for edge values. The downside it we won't be testing the formula, and so we'd need to amend test to check the adacent values (e.g. 3 and 262143).

Feel free to ignore.

Reply:
yeah, that is better!  Amended now